### PR TITLE
refactor!: update secret path to secretName

### PR DESCRIPTION
**Pull request by vli11**
BREAKING CHANGE: updated secret path to secretName

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions
make tidy
make test
make docker

### CODE REVIEW

**Review by jim-wang-yutsung — DISMISSED** [review]

File: internal/driver/credentials_test.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func TestTryGetCredentials(t *testing.T) {
 		mockAuthMode  string
 	}{
 		{
-			path:         noAuthSecretPath,
+			path:         noAuthSecretName,
```
Comment (lenny-goodell) [inline internal/driver/credentials_test.go]:
```suggestion
			secretName:         noAuthSecretName,
```

File: internal/driver/credentials_test.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func TestTryGetCredentialsForDevice(t *testing.T) {
 		existingProtocols map[string]models.ProtocolProperties
 		device            models.Device
 		expected          Credentials
-		path              string
+		name              string
```
Comment (lenny-goodell) [inline internal/driver/credentials_test.go]:
```suggestion
		secretName              string
```

**Review by lenny-goodell — CHANGES_REQUESTED** [review]

File: doc/credentials.md [hunk 3]
```diff
@@ -23,14 +23,14 @@ Three things must be done in order to add an authenticated camera to EdgeX:
 - **CredentialsMap**: (aka `AppCustom.CredentialsMap`) this contains the mappings between `Secret Path` and
     `MAC Address`. Each key in the map is a `Secret Path` which points to `Credentials` in the `Secret Store`. The value
     for each key is a comma separated list of `MAC Addresses` which should use those `Credentials`.
-- **DefaultSecretPath**: The `Secret Path` which points to the `Credentials` to use as the default for all devices
+- **DefaultSecretName**: The `Secret Path` which points to the `Credentials` to use as the default for all devices
     which are not configured in the `CredentialsMap`.
 - **NoAuth**: A special `Secret Path` that does not exist in the `Secret Store`. It is pre-configured as `Credentials`
```
Comment (jim-wang-yutsung) [inline doc/credentials.md]:
```suggestion
- **NoAuth**: A special `Secret Name` that does not exist in the `Secret Store`. It is pre-configured as `Credentials`
```

File: doc/custom-build.md [hunk 4]
```diff
@@ -68,7 +68,7 @@ Configuring pre-defined devices will allow the service to automatically provisio
 ### Configure the Device Service
 1. Open the [configuration.toml](./cmd/res/configuration.toml) file using your preferred text editor
 
-1. Make sure `path` is set to match `SecretPath` in `camera.toml`. In the sample below, it is `"credentials001"`. If you have multiple cameras, make sure the secret paths match.
+1. Make sure `path` is set to match `SecretName` in `camera.toml`. In the sample below, it is `"credentials001"`. If you have multiple cameras, make sure the secret paths match.
```
Comment (jim-wang-yutsung) [inline doc/custom-build.md]:
```suggestion
1. Make sure `secret name` is set to match `SecretName` in `camera.toml`. In the sample below, it is `"credentials001"`. If you have multiple cameras, make sure the secret names match.
```
Should `path` be secret name?

File: internal/driver/macmapper.go [hunk 5]
```diff
@@ -37,10 +38,10 @@ func (m *MACAddressMapper) UpdateMappings(raw map[string]string) {
...
 
 	credsMap := make(map[string]string)
-	for secretPath, macs := range raw {
-		if strings.ToLower(secretPath) != noAuthSecretPath { // do not check for noAuth
-			if _, err := m.sdkService.GetSecretProvider().GetSecret(secretPath, UsernameKey, PasswordKey, AuthModeKey); err != nil {
-				m.sdkService.GetLoggingClient().Warnf("One or more MAC address mappings exist for the secret path '%s' which does not exist in the Secret Store!", secretPath)
+	for secretName, macs := range raw {
+		if strings.ToLower(secretName) != noAuthSecretName { // do not check for noAuth
+			if _, err := m.sdkService.GetSecretProvider().GetSecret(secretName, UsernameKey, PasswordKey, AuthModeKey); err != nil {
+				m.sdkService.GetLoggingClient().Warnf("One or more MAC address mappings exist for the secret path '%s' which does not exist in the Secret Store!", secretName)
```
Comment (jim-wang-yutsung) [inline internal/driver/macmapper.go]:
```suggestion
				m.sdkService.GetLoggingClient().Warnf("One or more MAC address mappings exist for the secret name'%s' which does not exist in the Secret Store!", secretName)
```

File: cmd/res/configuration.toml [hunk 6]
```diff
@@ -37,7 +37,7 @@ RequestTimeout = 5 # Seconds
 # The Secret Path of the default credentials to use for devices which do not have MAC Addresses defined, or do not
```
Comment (jim-wang-yutsung) [inline cmd/res/configuration.toml]:
```suggestion
# The Secret Name of the default credentials to use for devices which do not have MAC Addresses defined, or do not
```

File: doc/credentials.md [hunk 7]
```diff
@@ -130,17 +130,17 @@ curl --location --request POST 'http://localhost:59984/api/v2/secret' \
 The device service supports three types of credential mapping. All three types can be used
 in conjunction with each other.
 
-- `1 to All` - All devices are given the default credentials based on `DefaultSecretPath`
+- `1 to All` - All devices are given the default credentials based on `DefaultSecretName`
 - `1 to Many` - In the `CredentialsMap`, one secret path can be assigned multiple MAC addresses
```
Comment (jim-wang-yutsung) [inline doc/credentials.md]:
```suggestion
- `1 to Many` - In the `CredentialsMap`, one secret name can be assigned multiple MAC addresses
```

File: doc/credentials.md [hunk 8]
```diff
@@ -130,17 +130,17 @@ curl --location --request POST 'http://localhost:59984/api/v2/secret' \
...
 - `1 to 1` - In the `CredentialsMap`, assign each secret path 1 MAC Address
```
Comment (jim-wang-yutsung) [inline doc/credentials.md]:
```suggestion
- `1 to 1` - In the `CredentialsMap`, assign each secret name 1 MAC Address
```

File: doc/custom-build.md [hunk 9]
```diff
@@ -68,7 +68,7 @@ Configuring pre-defined devices will allow the service to automatically provisio
...
 
 1. Under `path`, set `username` and `password` to your camera credentials. If you have multiple cameras copy the `Writable.InsecureSecrets` section and edit to include the new information.
```
Comment (jim-wang-yutsung) [inline doc/custom-build.md]:
```suggestion
1. Under `SecretName`, set `username` and `password` to your camera credentials. If you have multiple cameras copy the `Writable.InsecureSecrets` section and edit to include the new information.
```

File: doc/guides/CustomStartupGuide.md [hunk 10]
```diff
@@ -224,7 +224,7 @@ ONVIF devices support WS-Discovery, which is a mechanism that supports probing a
    # The target ethernet interface for multicast discovering
    DiscoveryEthernetInterface = "eth0"
    # The Secret Path of the default credentials to use for devices
```
Comment (jim-wang-yutsung) [inline doc/guides/CustomStartupGuide.md]:
```suggestion
   # The Secret Name of the default credentials to use for devices
```

File: doc/guides/CustomStartupGuide.md [hunk 11]
```diff
@@ -253,7 +253,7 @@ ONVIF devices support WS-Discovery, which is a mechanism that supports probing a
       # The target ethernet interface for multicast discovering
       APPCUSTOM_DISCOVERYETHERNETINTERFACE: "eth0"
       # The Secret Path of the default credentials to use for devices
```
Comment (jim-wang-yutsung) [inline doc/guides/CustomStartupGuide.md]:
```suggestion
      # The Secret Name of the default credentials to use for devices
```

File: doc/guides/CustomStartupGuide.md [hunk 12]
```diff
@@ -276,7 +276,7 @@ ONVIF devices support WS-Discovery, which is a mechanism that supports probing a
 
 2. Open the [configuration.toml](../../cmd/res/configuration.toml) file using your preferred text editor.
 
-3. Make sure `path` is set to match `SecretPath` in `camera.toml`. In the sample below, it is `"credentials001"`. If you have multiple cameras, make sure the secret paths match.
+3. Make sure `path` is set to match `SecretName` in `camera.toml`. In the sample below, it is `"credentials001"`. If you have multiple cameras, make sure the secret paths match.
```
Comment (jim-wang-yutsung) [inline doc/guides/CustomStartupGuide.md]:
```suggestion
3. Make sure `SecretName` is set to match `SecretName` in `camera.toml`. In the sample below, it is `"credentials001"`. If you have multiple cameras, make sure the secret names match.
```

File: doc/guides/CustomStartupGuide.md [hunk 13]
```diff
@@ -276,7 +276,7 @@ ONVIF devices support WS-Discovery, which is a mechanism that supports probing a
...
 
 4. Under `path`, set `username` and `password` to your camera credentials. If you have multiple cameras copy the `Writable.InsecureSecrets` section and edit to include the new information.
```
Comment (jim-wang-yutsung) [inline doc/guides/CustomStartupGuide.md]:
```suggestion
4. Under `SecretName`, set `username` and `password` to your camera credentials. If you have multiple cameras copy the `Writable.InsecureSecrets` section and edit to include the new information.
```

File: internal/driver/macmapper_test.go [hunk 14]
```diff
@@ -297,17 +297,17 @@ func TestTryGetSecretPathForMACAddress(t *testing.T) {
 		{
 			name:     "mac address for valid secret path",
```
Comment (jim-wang-yutsung) [inline internal/driver/macmapper_test.go]:
```suggestion
			name:     "mac address for valid secret name",
```

File: internal/driver/macmapper_test.go [hunk 15]
```diff
@@ -297,17 +297,17 @@ func TestTryGetSecretPathForMACAddress(t *testing.T) {
...
 			mac:      mappedMac,
-			expected: "valid_secret_path",
+			expected: "valid_secret_name",
 		},
 		{
 			name:     "mac address for default secret path",
```
Comment (jim-wang-yutsung) [inline internal/driver/macmapper_test.go]:
```suggestion
			name:     "mac address for default secret name",
```

File: internal/driver/macmapper_test.go [hunk 16]
```diff
@@ -283,11 +283,11 @@ func TestMACAddressMapper_UpdateMappings(t *testing.T) {
 	}
 }
 
-// TestTryGetSecretPathForMACAddress verifies the correct secret path is returned for a given mac address.
-func TestTryGetSecretPathForMACAddress(t *testing.T) {
+// TestTryGetSecretNameForMACAddress verifies the correct secret path is returned for a given mac address.
```
Comment (jim-wang-yutsung) [inline internal/driver/macmapper_test.go]:
```suggestion
// TestTryGetSecretNameForMACAddress verifies the correct secret name is returned for a given mac address.
```

File: internal/driver/macmapper.go [hunk 17]
```diff
@@ -52,36 +53,36 @@ func (m *MACAddressMapper) UpdateMappings(raw map[string]string) {
...
+		return noAuthSecretName
 	}
 
 	m.credsMu.RLock()
 	defer m.credsMu.RUnlock()
 
-	secretPath, found := m.credsMap[sanitized]
+	secretName, found := m.credsMap[sanitized]
 	if !found {
 		m.sdkService.GetLoggingClient().Debugf("No credential mapping exists for mac address '%s', will use default secret path.", mac)
```
Comment (jim-wang-yutsung) [inline internal/driver/macmapper.go]:
```suggestion
		m.sdkService.GetLoggingClient().Debugf("No credential mapping exists for mac address '%s', will use default secret name.", mac)
```

File: internal/driver/macmapper.go [hunk 18]
```diff
@@ -52,36 +53,36 @@ func (m *MACAddressMapper) UpdateMappings(raw map[string]string) {
...
+				credsMap[sanitized] = secretName
 			}
 		}
 	}
 
 	m.credsMap = credsMap
 }
 
-// TryGetSecretPathForMACAddress will return the secret path associated with the mac address passed if a mapping exists,
+// TryGetSecretNameForMACAddress will return the secret path associated with the mac address passed if a mapping exists,
```
Comment (jim-wang-yutsung) [inline internal/driver/macmapper.go]:
```suggestion
// TryGetSecretNameForMACAddress will return the secret name associated with the mac address passed if a mapping exists,
```

**Review by jim-wang-yutsung — CHANGES_REQUESTED** [review]

**Review by lenny-goodell — CHANGES_REQUESTED** [review]
the scripts in bin folder also need updating.

File: doc/auto-discovery.md [hunk 19]
```diff
@@ -40,7 +40,7 @@ Define the following configurations in [cmd/res/configuration.toml](../cmd/res/c
 # The target ethernet interface for multicast discovering
 DiscoveryEthernetInterface = "eth0"
 # The Secret Path of the default credentials to use for devices
```
Comment (jim-wang-yutsung) [inline doc/auto-discovery.md]:
```suggestion
# The Secret Name of the default credentials to use for devices
```

File: doc/auto-discovery.md [hunk 20]
```diff
@@ -62,7 +62,7 @@ device-onvif-camera:
     # The target ethernet interface for multicast discovering
     APPCUSTOM_DISCOVERYETHERNETINTERFACE: "eth0"
     # The Secret Path of the default credentials to use for devices
```
Comment (jim-wang-yutsung) [inline doc/auto-discovery.md]:
```suggestion
    # The Secret Name of the default credentials to use for devices
```

File: internal/driver/credentials.go [hunk 21]
```diff
@@ -87,23 +87,23 @@ func (d *Driver) tryGetCredentials(secretPath string) (Credentials, errors.EdgeX
...
+		secretName = d.macAddressMapper.TryGetSecretNameForMACAddress(mac, defaultSecretName)
 	} else {
 		d.lc.Warnf("Device %s is missing MAC Address, using default secret path", device.Name)
 	}
 
-	credentials, edgexErr := d.tryGetCredentials(secretPath)
+	credentials, edgexErr := d.tryGetCredentials(secretName)
 	if edgexErr != nil {
-		d.lc.Errorf("Failed to retrieve credentials for the secret path %s: %s", secretPath, edgexErr.Error())
+		d.lc.Errorf("Failed to retrieve credentials for the secret path %s: %s", secretName, edgexErr.Error())
```
Comment (jim-wang-yutsung) [inline internal/driver/credentials.go]:
```suggestion
		d.lc.Errorf("Failed to retrieve credentials for the secret name %s: %s", secretName, edgexErr.Error())
```

File: internal/driver/credentials.go [hunk 22]
```diff
@@ -87,23 +87,23 @@ func (d *Driver) tryGetCredentials(secretPath string) (Credentials, errors.EdgeX
...
 		return Credentials{}, errors.NewCommonEdgeX(errors.KindServerError, "failed to get credentials", edgexErr)
 	}
 
-	d.lc.Debugf("Found credentials from secret path %s for device %s", secretPath, device.Name)
+	d.lc.Debugf("Found credentials from secret path %s for device %s", secretName, device.Name)
```
Comment (jim-wang-yutsung) [inline internal/driver/credentials.go]:
```suggestion
	d.lc.Debugf("Found credentials from secret name %s for device %s", secretName, device.Name)
```

**Review by jim-wang-yutsung — DISMISSED** [review]
LGTM

File: internal/driver/macmapper.go [hunk 23]
```diff
@@ -7,16 +7,17 @@
 package driver
```
Comment (ChristianDarr-personal) [inline internal/driver/macmapper.go:7]: There are a few comments missed in this file
```
//UpdateMappings takes the raw map of secret path to csv list of mac addresses and
// inverts it into a quick lookup map of mac address to secret path.
// the default secret path if the mapping is not found, or no auth if the mac address is invalid.
// if the secret path is the special NoAuth magic key, do not look it up, instead return the noAuthCredentials
// the default secret path will be used to look up the credentials. An error is returned if the secret path
```

File: internal/driver/config.go [hunk 24]
```diff
@@ -17,8 +17,8 @@ import (
 type CustomConfig struct {
```
Comment (ChristianDarr-personal) [inline internal/driver/config.go:17]: On line 20, 
// DefaultSecretName indicates the secret `path` to retrieve username and password from secret store.

**Review by ChristianDarr-personal — CHANGES_REQUESTED** [review]
In [doc/utility-scripts.md](doc/utility-scripts.md) a the secret path arguments need changed.
```
bin/edit-credentials.sh [-s/--secure-mode] [-u <username>] [-p <password>] [--auth-mode {usernametoken|digest|both}] `[-P secret-path]` [-M mac-addresses] [-t <consul token>]
bin/map-credentials.sh [-s/--secure-mode] [-u <username>] [-p <password>] [--auth-mode {usernametoken|digest|both}] `[-P secret-path]` [-M mac-addresses] [-t <consul token>]
bin/query-mappings.sh [-s/--secure-mode] [-u <username>] [-p <password>] [--auth-mode {usernametoken|digest|both}] `[-P secret-path]` [-M mac-addresses] [-t <consul token>]
```

I discovered some of these using `grep -R "path" *`

File: internal/driver/macmapper.go (hunk 23 above)
Comment (vli11) [reply internal/driver/macmapper.go:7]: I can update the comments, and update secret path to secret name, but it looks like the comment is not completed, can you provide the missing comments?

Comment (vli11) [reply internal/driver/macmapper.go:7]: or @lenny-intel can you help to complete the comments?

Comment (lenny-goodell) [reply internal/driver/macmapper.go:7]: @vli11 , that is just an artifact of how @ChristianDarr-personal is showing the missed updates. 
All you need to do is search for `path` in the file and update the missed references to `path`

Comment (ChristianDarr-personal) [reply internal/driver/macmapper.go:7]: My bad, some of the comments got chopped off - I was just hoping to show the comments that had `path` in them

**Review by lenny-goodell — COMMENTED** [review]
LGTM, pending validation

File: doc/onvif-user-authentication.md [hunk 25]
```diff
@@ -31,7 +31,7 @@ The AuthMode can be:
 * both
 * none
 
-SecretPath should contain:
+SecretName should contain:
 * username
 * password
```
Comment (ChristianDarr-personal) [inline doc/onvif-user-authentication.md:37]: Lines 44 and 50 use the word path in the configuration.toml

**Review by ChristianDarr-personal — DISMISSED** [review]
Line 57 in `using-demo-app.md` uses path

File: doc/onvif-user-authentication.md (hunk 25 above)
Comment (lenny-goodell) [reply doc/onvif-user-authentication.md:37]: @vli11 , @ChristianDarr-personal is referring to the in-line TOML snippet below, not the actual configuration file.

**Comment by vchitike** [comment]
Observed below two issues during validation:
1.Using map_credentials.sh script it adds credentials at writable/insecuresecrets/secretname/secrets.
 authorization does not work by adding credentials at above path. I tried manually adding credentials in consul at writable/insecuresecrets/secrets/secretdata. With this authorization does work correctly for camera(s). All utility scripts (map_credentials, edit_credentials etc) need to be updated with 'SecretName' instead of secretpath.
2. after adding credentials, I have to restart onvif service for authorization work. This needs to be investigated and resolved.

**Review by lenny-goodell — COMMENTED** [review]
LGTM pending validation

**Comment by jim-wang-yutsung** [comment]
the script bin/utils.sh: https://github.com/edgexfoundry/device-onvif-camera/blob/main/bin/utils.sh#L505 line has json path. Question: is this json data ok?

**Review by jim-wang-yutsung — COMMENTED** [review]
Other than validation, changes LGTM

**Comment by vchitike** [comment]
Validation still does not look good. 
Not authorizing cameras until user restarts onvif service. also after its authorizing it did not add at least one authorized camera to metadata successfully. attached logs.
[_edgex-device-onvif-camera_logs (47).txt](https://github.com/edgexfoundry/device-onvif-camera/files/10887129/_edgex-device-onvif-camera_logs.47.txt)

**Comment by vli11** [comment]
> Validation still does not look good. Not authorizing cameras until user restarts onvif service. also after its authorizing it did not add at least one authorized camera to metadata successfully. attached logs. [_edgex-device-onvif-camera_logs (47).txt](https://github.com/edgexfoundry/device-onvif-camera/files/10887129/_edgex-device-onvif-camera_logs.47.txt)

thanks for testing @vyshali-chitikeshi , can you share steps how you get to this error, do you need real camera to test it?

**Comment by lenny-goodell** [comment]
@vyshali-chitikeshi , the `request failed, status code: 503, err: HTTP request timeout"` errors are from the know issue, which I have a fix for going in today. The fix is in Core Metadata's Add Device. I don't see any errors accessing the secrets, which this PR impacts with the renaming of fields. Lets retest once my Core Metadata fix is ready and see if that helps clear up other issue also.

File: bin/utils.sh [hunk 26]
```diff
@@ -492,17 +492,17 @@ parse_args() {
...
+    put_insecure_secrets_field "${SECRET_NAME}/SecretData/password"    "${SECRET_PASSWORD}"
+    put_insecure_secrets_field "${SECRET_NAME}/SecretData/mode"        "${AUTH_MODE}"
 }
 
 # set the secure secrets by posting to the device service's secret endpoint
 set_secure_secret() {
     local payload="{
     \"apiVersion\":\"v2\",
-    \"path\": \"${SECRET_PATH}\",
+    \"path\": \"${SECRET_NAME}\",
```
Comment (ajcasagrande) [inline bin/utils.sh:505]: @vli11 does this need to be changed from `"path"`?

**Review by lenny-goodell — APPROVED** [review]
LGTM and passed validation

File: bin/utils.sh (hunk 26 above)
Comment (lenny-goodell) [reply bin/utils.sh:505]: Yes.

**Pull request merged**